kong.conf.default

```diff
@@ -80,15 +80,25 @@
                                  # `bundled` and plugin names can be mixed
                                  # together, as the following examples suggest:
                                  #
-                                 # - plugins = bundled, custom-auth
-                                 #     > will load all bundled plugins and
-                                 #       another plugin `custom-auth`
+                                 # * `plugins=bundled,custom-auth,custom-log` will
+                                 #   include the bundled plugins plus two custom ones
+                                 # * `plugins=custom-auth,custom-log` will *only*
+                                 #    include the `custom-auth` and `custom-log` plugins.
+                                 # * `plugins=off` will not include any plugins
                                  #
-                                 # - plugins = key-auth, custom-auth
-                                 #     > will only load these two plugins
                                  #
-                                 # - plugins = off
-                                 #     > will not load any plugin
+                                 #  **Note:** Kong will not start if some plugins
+                                 #  were previously configured (i.e. have rows in
+                                 #  the database) and are not specified in this list.
+                                 #  Before disabling a plugin, ensure all instances
+                                 #  of it are removed before restarting Kong.
+                                 #
+                                 #
+                                 #  **Note:** Limiting the amount of available plugins
+                                 #  can improve P99 latency when experiencing LRU churning
+                                 #  in the database cache (i.e. when the configured
+                                 #  [mem_cache_size](#mem_cache_size) is full.
```

Comment:
This link will not lead anywhere in this file. I am starting to have doubts regarding this whole endeavor. Is the goal to serve the cleanest configuration file to our users, or to simplify our own development process wrt. maintaining the docs...? Yes, automated docs makes for a more consistent experience (reduces the chances of said docs being out of date), but the proper way to go would probably be to write a script that automatically transforms all such references of other config variables into markdown links. And not polluting this file with Markdown format...

Reply:
Totally agreed. It was an oversight.

Yes, the purpose of this PR is facilitating the path towards automating the doc generation. But I wanted to get familiar with the docs (doing a "manual sync") before attempting to write an automated tool.

I am now more capable of doing that other task, having gone through the docs manually. The automation will be simpler since the docs have less differences now.